pulp_rpm/tests/functional/api/test_domains.py

```diff
@@ -0,0 +1,288 @@
+
+from pulp_smash.pulp3.utils import gen_repo
+
+from pulp_rpm.tests.functional.utils import (
+    gen_rpm_remote,
+    get_package_repo_path,
+)
+from pulp_rpm.tests.functional.constants import (
+    RPM_SIGNED_FIXTURE_URL,
+)
+
+if not settings.DOMAIN_ENABLED:
+    pytest.skip("Domains not enabled.", allow_module_level=True)
+
+
+def test_domain_create(
+    domains_api_client,
+    gen_object_with_cleanup,
+    monitor_task,
+    rpm_repository_api,
+    rpm_rpmremote_api,
+):
+    """Test domain creation."""
+    new_domain = str(uuid.uuid4())
+    body = {
+        "name": new_domain,
+        "storage_class": "pulpcore.app.models.storage.FileSystem",
+        "storage_settings": {"MEDIA_ROOT": "/var/lib/pulp/media/"},
+    }
+    domain = gen_object_with_cleanup(domains_api_client, body)
+    domain_name = domain.name
+    assert domain_name == new_domain
+
+    # sync remote in default domain (not specified)
+    remote = gen_object_with_cleanup(rpm_rpmremote_api, gen_rpm_remote(RPM_SIGNED_FIXTURE_URL))
+    repo = gen_object_with_cleanup(rpm_repository_api, gen_repo(remote=remote.pulp_href))
+    sync_url = RpmRepositorySyncURL(remote=remote.pulp_href)
+    sync_response = rpm_repository_api.sync(repo.pulp_href, sync_url)
+    monitor_task(sync_response.task)
+
+    assert rpm_repository_api.list(pulp_domain="default").count == 1
+    # check that newly created domain doesn't have any repo
+    assert rpm_repository_api.list(pulp_domain=new_domain).count == 0
+
+
+@pytest.mark.parallel
+def test_object_creation(
+    domains_api_client, gen_object_with_cleanup, rpm_repository_api, rpm_rpmremote_api
+):
+    """Test basic object creation in a separate domain."""
+    body = {
+        "name": str(uuid.uuid4()),
+        "storage_class": "pulpcore.app.models.storage.FileSystem",
+        "storage_settings": {"MEDIA_ROOT": "/var/lib/pulp/media/"},
+    }
+    domain = gen_object_with_cleanup(domains_api_client, body)
+    domain_name = domain.name
+
+    repo_body = {"name": str(uuid.uuid4())}
+    repo = gen_object_with_cleanup(rpm_repository_api, repo_body, pulp_domain=domain_name)
+    assert f"{domain_name}/api/v3/" in repo.pulp_href
+
+    repos = rpm_repository_api.list(pulp_domain=domain_name)
+    assert repos.count == 1
+    assert repo.pulp_href == repos.results[0].pulp_href
+
+    # Will list repos on default domain
+    default_repos = rpm_repository_api.list(name=repo.name)
+    assert default_repos.count == 0
+
+    # Try to create an object w/ cross domain relations
+    default_remote = gen_object_with_cleanup(rpm_rpmremote_api, gen_rpm_remote())
+    with pytest.raises(ApiException) as e:
+        repo_body = {"name": str(uuid.uuid4()), "remote": default_remote.pulp_href}
+        rpm_repository_api.create(repo_body, pulp_domain=domain.name)
+    assert e.value.status == 400
+    # What key should this error be under? non-field-errors seems wrong
```

Comment:
Hmm good point, but as it is checking all the fields + where the request is being made from I didn't see any good keys to put this under. We could change it in pulpcore.